ouroboros-consensus/test-consensus/Test/ThreadNet/General.hs

```diff
@@ -594,3 +598,18 @@ prop_general countTxs k TestConfig{numSlots, nodeJoinPlan, nodeRestarts, nodeTop
         average :: [Double] -> Double
         average [] = 0
         average xs = sum xs / fromIntegral (length xs)
+
+    -- The 'prop_valid_block' argument could, for example, check for no expired
+    -- transactions.
+    --
+    -- TODO apply to all 'TraceForgeAdopted' blocks, not just those in the
+    -- final chain?
```

Comment:
Is that possible? Maybe by (ab?)using the tracer you have for `TraceForgeAdopted`?

Reply:
Yeah, I was thinking via tracer. It doesn't seem like an abuse, I don't think: the tracer could just accumulate the adopted blocks in a set and check them at the end.